.github/workflows/_python-vulnerability-scan.yml

```diff
@@ -5,7 +5,7 @@ on:
 
 jobs:
   bandit-scan:
-    runs-on: ubuntu-latest
+    runs-on: blacksmith-4vcpu-ubuntu-2204
```

Comment:
I wonder if we should use the `blacksmith-2vcpu-ubuntu-2204` (aka just `blacksmith`) for these "smaller" jobs?

Reply:
In this PR I am trying to keep as much parity with the existing setup as possible.

The default -latest runners on GH are 4vcpu - https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources

Reply:
We can perform such optimizations in the scope of another change.